### PR TITLE
refactor: payment methods list

### PR DESCRIPTION
**Pull request by frosso**
# Changes proposed in this Pull Request:

We need a slight refactor to the payment methods list.
- Added a loadable section, since I was on it
- Added a "Get more payment methods" button
- The "Pending activation" pill is now clickable
- The elements on the list no longer control the checkbox status (confirmed with UX - otherwise we wouldn't be able to make the tooltip clickable)

With UPE disabled:
Screen Shot 2021-09-16 at 2 31 55 PM
Screen Shot 2021-09-16 at 2 32 09 PM

With UPE enabled:
Screen Shot 2021-09-16 at 2 33 26 PM
Screen Shot 2021-09-16 at 2 32 48 PM

Tooltip:
Screen Shot 2021-09-16 at 2 39 57 PM

# Testing instructions

- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe
- The list is now updated
- The "get more payment methods" button links to Stripe
- The "Pending activation" pill shows additional information
- Responsive behavior of the list hasn't changed (elements still wrap in the same way as before)

### CODE REVIEW

File: client/components/tooltip/index.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ const Tooltip = ( { isVisible, onHide = noop, ...props } ) => {
 		setIsHovered( false );
 		onHide();
 	};
-	const handleMouseClick = () => {
+	const handleMouseClick = ( event ) => {
+		event.preventDefault();
```
Comment (frosso) [inline client/components/tooltip/index.js:17]: Prevents navigating to another page when clicking on the pill

File: client/settings/card-body.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { CardBody } from '@wordpress/components';
 export default styled( CardBody )`
 	// increasing the specificity of the styles to override the Gutenberg ones
 	&.is-size-medium.is-size-medium {
-		padding: $grid-unit-30;
+		padding: 24px;
```
Comment (frosso) [inline client/settings/card-body.js:7]: Fixed on all components 😬

File: client/settings/general-settings-section/index.js [hunk 3]
```diff
@@ -3,148 +3,30 @@
...
-import PaymentMethodSetupHelp from './payment-method-setup-help';
+import SectionFooter from './section-footer';
+import PaymentMethodsList from './payment-methods-list';
+import UpeToggleContext from 'wcstripe/settings/upe-toggle/context';
 
 const StyledCard = styled( Card )`
 	margin-bottom: 12px;
 `;
 
-const PaymentMethodsList = styled.ul`
```
Comment (frosso) [inline client/settings/general-settings-section/index.js:20]: All of this has been extracted to the `PaymentMethodsList` component in `./payment-methods-list`

File: client/settings/general-settings-section/payment-methods-list.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+/**
```
Comment (frosso) [inline client/settings/general-settings-section/payment-methods-list.js]: Extracted from `client/settings/general-settings-section/index.js`

File: client/settings/general-settings-section/section-heading.js [hunk 5]
```diff
@@ -43,16 +42,10 @@ const Title = styled.h4`
 `;
 
 const SectionHeading = () => {
-	const { isUpeEnabled } = useContext( UpeToggleContext );
```
Comment (frosso) [inline client/settings/general-settings-section/section-heading.js:39]: Moved to `client/settings/general-settings-section/index.js` (already covered by tests)

**Comment by harriswong** [comment]
👀

**Review by harriswong — APPROVED** [review]
Verified `wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe` has pill tooltip and  "Get more payment methods" button. LGTM
👍

**Pull request merged**